MouseMovementTool.cs

```diff
@@ -34,87 +27,71 @@ public Transform cameraRig
-            }
-        }
-
-        //public void ProcessInput(ActionMapInput input, ConsumeControlDelegate consumeControl)
-        //{
-        //    throw new NotImplementedException();
-        //}
-
-        public GameObject instructions;
-        public float movementMultiplier = .1f;
-        public GameObject ringPrefab;
-        public Ring ring;
-
-        private void ToggleInstructions()
-        {
-            if(instructions)
-                instructions.SetActive(!instructions.activeSelf);
-        }
-
-        private void Start()
-        {
-            GameObject instance = Instantiate(ringPrefab, cameraRig, false);
+            GameObject instance = Instantiate(m_ringPrefab, cameraRig, false);
             instance.transform.localPosition = new Vector3(0f, -0.09f, 0.18f);
-            ring = instance.GetComponent<Ring>();
+            m_ring = instance.GetComponent<Ring>();
         }
 
-        private void FixedUpdate()
+        void FixedUpdate()
         {
+            bool k_mouse0 = UnityEngine.Input.GetMouseButton(0);
+            bool k_mouse0down = UnityEngine.Input.GetMouseButtonDown(0);
+            bool k_mouse1 = UnityEngine.Input.GetMouseButton(1);
+            bool k_mouse1down = UnityEngine.Input.GetMouseButtonDown(1);
 
-            if (UnityEngine.Input.GetMouseButtonDown(0) && UnityEngine.Input.GetMouseButtonDown(1) ||
-           UnityEngine.Input.GetMouseButton(0) && UnityEngine.Input.GetMouseButtonDown(1) ||
-           UnityEngine.Input.GetMouseButtonDown(0) && UnityEngine.Input.GetMouseButton(1))
+            if (k_mouse0down && k_mouse1down ||
+           k_mouse0 && k_mouse1down ||
+           k_mouse0down && k_mouse1)
             {
                 ToggleInstructions();
             }
-            if (UnityEngine.Input.GetMouseButton(0))
+            if (k_mouse0)
             {
-                Vector3 forward = Vector3.Scale(cameraRig.forward, new Vector3(1f, 0f, 1f)).normalized;
-                Vector3 right = Vector3.Scale(cameraRig.right, new Vector3(1f, 0f, 1f)).normalized;
-                Vector3 delta = (UnityEngine.Input.GetAxis("Mouse X") * right + UnityEngine.Input.GetAxis("Mouse Y") * forward) * movementMultiplier;
-                //Debug.Log(delta.normalized);
-                cameraRig.position += delta;
+                const string k_mouseX = "Mouse X";
+                const string k_mouseY = "Mouse Y";
 
-                //Update trackball
-                //ball.rotation = Quaternion.AngleAxis(-UnityEngine.Input.GetAxis("Mouse X") * movementMultiplier * 10f, trackball.forward)
-                //    * Quaternion.AngleAxis(UnityEngine.Input.GetAxis("Mouse Y") * movementMultiplier * 10f, trackball.right) * ball.rotation;
+                Vector3 k_forward = Vector3.Scale(cameraRig.forward, new Vector3(1f, 0f, 1f)).normalized;
+                Vector3 k_right = Vector3.Scale(cameraRig.right, new Vector3(1f, 0f, 1f)).normalized;
```

Comment:
you should be able to calculate right as the perpendicular version of forward -k_forward.z, 0, k_forward.x which will handle the case of the camera rig being rotated along the roll axis more gracefully